In [82]:
import pandas as pd 
import numpy as np

Paulina González Ramírez 81678 Práctica 1.1

In [83]:
df = pd.read_csv('../data/raw/annual-population-growth/annual-population-growth.csv')

In [84]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [85]:
df.shape

(38400, 5)

1.	¿Cuántas filas y columnas tiene el dataset?
38400 filas y 5 columnas

In [86]:
df.columns

Index(['Entity', 'Code', 'Year', 'Annual change in population',
       'Annual population change (Projected)'],
      dtype='str')

In [87]:
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [88]:
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

2.	¿Qué tipo de datos contiene la variable Year?
entero, int 64

In [89]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [90]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

3.	¿Existen valores faltantes?
si entity y year

In [91]:
df["Entity"].nunique()

256

4.	¿Cuántas entidades diferentes contiene el dataset? 256

In [92]:
anio_consultado= 2020

In [93]:
print("año consultado:", anio_consultado)

año consultado: 2020


In [94]:
df_anio_consultado = df[df["Year"]==anio_consultado & (df["Entity"]!="Venezuela")]
df_anio_consultado.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)


In [95]:
df_anio_consultado.mean(numeric_only=True)

Year                                   NaN
Annual change in population            NaN
Annual population change (Projected)   NaN
dtype: float64

In [96]:
df_anio_consultado.shape

(0, 5)

In [97]:
df.head(3)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN


In [98]:
df_year=df[["Entity","Year"]]
df_year.head()

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955


In [99]:
df_extraccion_filas = df[df["Year"] == 2020]
df_extraccion_filas.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN
519,Algeria,DZA,2020,747543.0,NaN
669,American Samoa,ASM,2020,-451.0,NaN


In [100]:
df_2020_mex = df [(df["Year"] == 2020) & (df["Code"] == "MEX")]
df_2020_mex.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [101]:
#Rango de fechas desdeñaño que nacimos hasta el 2020
df_2005_2020_mex = df [(df["Year"] >= 2005) & (df["Year"] <= 2020) & (df["Code"] == "MEX")]
df_2005_2020_mex.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22254,Mexico,MEX,2005,1417373.0,NaN
22255,Mexico,MEX,2006,1442162.0,NaN
22256,Mexico,MEX,2007,1520690.0,NaN
22257,Mexico,MEX,2008,1599930.0,NaN
22258,Mexico,MEX,2009,1625434.0,NaN


In [102]:
df_ext_EYA = df_2005_2020_mex[["Entity","Year","Annual change in population"]]
df_ext_EYA.head()

,Entity,Year,Annual change in population
22254,Mexico,2005,1417373.0
22255,Mexico,2006,1442162.0
22256,Mexico,2007,1520690.0
22257,Mexico,2008,1599930.0
22258,Mexico,2009,1625434.0


In [103]:
#Promedio del rango esntre el 2005 y 2020 en México de personas 
df_ext_EYA.mean(numeric_only=True)

Year                              2012.500
Annual change in population    1400307.625
dtype: float64

In [104]:
df.loc[:,["Entity", "Year"]]

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955
...,...,...
38395,Zimbabwe,2096
38396,Zimbabwe,2097
38397,Zimbabwe,2098
38398,Zimbabwe,2099


In [105]:
df.iloc[:,0:2]

,Entity,Code
0,Afghanistan,AFG
1,Afghanistan,AFG
2,Afghanistan,AFG
3,Afghanistan,AFG
4,Afghanistan,AFG
...,...,...
38395,Zimbabwe,ZWE
38396,Zimbabwe,ZWE
38397,Zimbabwe,ZWE
38398,Zimbabwe,ZWE


In [106]:
df.iloc[30:35,0:12]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
30,Afghanistan,AFG,1981,-1231727.0,NaN
31,Afghanistan,AFG,1982,-946206.0,NaN
32,Afghanistan,AFG,1983,-73395.0,NaN
33,Afghanistan,AFG,1984,272237.0,NaN
34,Afghanistan,AFG,1985,236633.0,NaN


In [107]:
df.iloc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [108]:
#ej2
df.loc[:,["Entity", "Year", "Annual change in population"]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [109]:
#ej3
df.loc[10000,["Entity","Year","Annual change in population"]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [110]:
#ej4
df.loc[(df["Year"] == 2020) & (df["Entity"] == "Mexico")]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [111]:
#ej5
df.iloc[:5]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [112]:
#ej6
df.iloc[["0","2","3"]]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN


In [113]:
#ej7
df.iloc[0:4,0:4]

,Entity,Code,Year,Annual change in population
0,Afghanistan,AFG,1951,103163.0
1,Afghanistan,AFG,1952,108441.0
2,Afghanistan,AFG,1953,108919.0
3,Afghanistan,AFG,1954,111251.0


In [114]:
#ejercicio con año de nacimiento e inicial de mi apellido en un país
df_filtrado=df[(df["Year"]>= 2005) &(df["Year"]<=2023) & (df["Entity"] == "Equatorial Guinea")][["Annual change in population"]]
df_filtrado.head()

,Annual change in population
9954,47446.0
9955,42979.0
9956,45704.0
9957,48485.0
9958,51145.0


In [115]:
print(df_filtrado.mean())

Annual change in population    49875.368421
dtype: float64


In [116]:
print(df_filtrado.median())

Annual change in population    51642.0
dtype: float64


In [117]:
print(df_filtrado.var())

Annual change in population    1.884482e+07
dtype: float64


In [118]:
print(df_filtrado.std())

Annual change in population    4341.06165
dtype: float64


In [119]:
print(df_filtrado.min())

Annual change in population    42317.0
dtype: float64


In [120]:
print(df_filtrado.max())

Annual change in population    54267.0
dtype: float64


Preguntas actividad 3:
1.	¿Cuál es el valor de la media?
49875.368421
2.	¿Cuál es el valor de la mediana?
 51642.0
3.	¿Cuál es el valor de la desviación estándar?
4341.06165
4.	¿Qué entidad presenta el valor mínimo y cuál el máximo?
42317.0 y 54267.0

In [121]:

import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.14.6 (tags/v3.14.6:c63aec6, Jun 10 2026, 10:26:10) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


prueba de git jiji